src/MoBi.Core/Services/SimulationLoader.cs

```diff
@@ -68,14 +72,26 @@ private void addSimulationToProject(IMoBiSimulation simulation, MoBiMacroCommand
 
          addSimulationConfigurationToProject(moBiSimulation, loadCommand);
 
-         moBiSimulation.ResultsDataRepository = simulation.ResultsDataRepository;
-         if (!_nameCorrector.CorrectName(project.Simulations, moBiSimulation))
-            return;
-
          moBiSimulation.HasChanged = true;
          loadCommand.AddCommand(new AddSimulationCommand(moBiSimulation));
       }
 
+      private void renameCollidingModules(IEnumerable<Module> modulesToRename, string simulationName, IReadOnlyList<IWithName> existingEntities)
+      {
+         var takenNames = existingEntities.AllNames();
+
+         modulesToRename
+            .Where(x => takenNames.Contains(x.Name))
+            .Select((x, index) => new { Module = x, Index = index })
+            .Each(item => item.Module.Name = getModuleNameByIndex(simulationName,item.Index) );
+
+         //If there are still colliding names based on the simulation name, we need to correct them
+         modulesToRename.Where(x => takenNames.Contains(x.Name)).Each(x => _nameCorrector.AutoCorrectName(takenNames, x));
+      }
+
+      private string getModuleNameByIndex(string name, int index)=>
+         $"{name}{(index == 0 ? string.Empty : " " + index)}";
```

Comment:
You can probably use the `nameCorrector.AutoCorrectName` which does the index generation for you. I think you could get rid of the `Select` above and just do an `Each` and it should be simple enough that you don't need this method then.

Reply:
ah, no, that won't work with Autocorrect because it just enumerates the current name, you actually want to rename and index in one step